pkgs/applications/misc/seatd/default.nix

```diff
@@ -25,9 +25,11 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ meson ninja pkg-config scdoc ];
 
-  buildInputs = [ systemd ];
+  buildInputs = lib.optionals systemdSupport [ systemd ];
 
-  mesonFlags = [ "-Dlibseat-logind=systemd" "-Dlibseat-builtin=enabled" ];
+  mesonFlags = if systemdSupport
+               then [ "-Dlibseat-logind=systemd" "-Dlibseat-builtin=enabled" ]
+               else [ "-Dserver=enabled" "-Dseatd=enabled" "-Dlogind=disabled" "-Dauto_features=auto" ];
```

Comment:
Do we really need a completely different set of build flags just to toggle systemd support?

Reply:
Sorry about that, let me clean it up.

Reply:
Please see newly-pushed patch.

I've been carrying this patch for a while (through the libseat->seatd rename).  It looks like at some point seatd renamed their options, and meson silently allows options it doesn't understand (ugh).  I pruned the inert options.